### ISSUE TITLE
Question about reproducing NTv3 post-finetuning species genome data splits (other species / homology-based method)

### ISSUE THREAD

**Issue by Liujh0223** [instadeepai/nucleotide-transformer#132]
Hi NTv3 team,

Thank you for open-sourcing NTv3 and the associated resources. After carefully reading the paper, I learned a lot from the design choices and experiments.

While going through the post-finetuning section, I noticed that the “Species Genome Data Splits” generation—especially for “Other Species”—uses a homology-based strategy (Section 4.6.3, p.41). I’m currently trying to reproduce the data split procedure for rice based on the description in the paper, but my results are not matching expectations.

Could you please share guidance on how to implement the splitting method described there in practice? If there are any more detailed step-by-step instructions, scripts, or an open-source reference implementation (even partial) that correspond to the procedure in Section 4.6.3, I would really appreciate it.

Thanks again for your work and for making the model publicly available.

Best regards,
Jianhong Liu.

<img width="1686" height="1269" alt="Image" src="https://github.com/user-attachments/assets/4f52e4e9-9944-48e8-9780-800a2e5b98d5" />

**Comment by bernardo-de-almeida** [comment]
Hi @Liujh0223 , thanks for the question. We have shared the data splits of the different species on HuggingFace - here are the rice ones: https://huggingface.co/datasets/InstaDeepAI/NTv3_benchmark_dataset/blob/main/rice/splits.bed

Is that information enough?

**Comment by Liujh0223** [comment]
Thank you for your reply. I noticed the rice/splits.bed file in the NTv3 benchmark dataset (https://huggingface.co/datasets/InstaDeepAI/NTv3_benchmark_dataset/blob/main/rice/splits.bed
), but I’m not entirely sure how these regions were defined. I tried to retrieve rice homoeolog/ortholog gene pairs from Ensembl Plants / BioMart and then assessed homology between regions according to the provided splits.bed. However, I still found homologous genes shared across the test–val and test–train regions. My guess is that this may be due to threshold/boundary effects (e.g., genes near region boundaries) or that the clustering/partitioning was not fully effective.

**Comment by bernardo-de-almeida** [comment]
Exactly, you are right. Let me clarify how the splits were generated:

The homology-based clustering approach we used does not completely eliminate homologous genes across train/validation/test splits, rather it minimizes such data leakage by pruning the weakest homology connections.

Here are the main steps:

1. Edge weighting: The connectivity matrix between genomic windows is weighted by the sum of base pair sizes of shared homologous gene pairs. This means stronger homology relationships (larger shared gene content) create stronger edges between windows.

2. MCL clustering: The Markov Clustering (MCL) algorithm uses these weighted edges to create clusters by iteratively pruning weaker connections. However, MCL doesn't guarantee zero homology between clusters - it creates clusters where intra-cluster homology is much stronger than inter-cluster homology. Some residual homology across clusters is expected.

3. Rice-specific pruning: For rice (O. sativa), as mentioned in the paper, we applied additional edge pruning. Specifically, edges representing shared homology of less than 10% of the window size were removed before clustering. This was necessary to achieve our predefined validation and test set sizes while maintaining reasonable cluster cohesion.

The goal was to create splits that substantially reduce data leakage from strong homology relationships while maintaining biologically meaningful and appropriately sized test/validation sets, rather than achieving complete homology isolation (which would be overly restrictive for many species).

Does it make sense?

**Comment by Liujh0223** [comment]
Thank you for the clear explanation — it really made the splitting strategy click for me.

My understanding is that the key is not eliminating homologous genes entirely, but treating windows as the unit of assignment and using homologous gene pairs between windows to quantify inter-window connectivity (edge weights), so that windows with strong homology are grouped into the same split.

It’s a very elegant approach and definitely worth learning from. Thanks again for your help🌹🌹🌹.

Best regards,
Jianhong Liu.
